aspnetcore/mvc/models/validation.md

```diff
@@ -76,13 +76,19 @@ Client-side validation requires a value for a form field that corresponds to a m
 
 Model state represents validation errors in submitted HTML form values.
 
-MVC will continue validating fields until reaches the maximum number of errors (200 by default). You can configure this number by inserting the following code into the `ConfigureServices` method in the *Startup.cs* file:
+MVC will continue validating fields until it reaches the maximum number of errors (200 by default). You can configure this number with the following code in `Startup.ConfigureServices`:
 
 [!code-csharp[](validation/sample/Startup.cs?range=27)]
 
-## Handling Model State Errors
+## Handle Model State errors
+
+Model validation occurs before the execution of a controller action. It's the action's responsibility to inspect `ModelState.IsValid` and react appropriately. In many cases, the appropriate reaction is to return an error response, ideally detailing the reason why model validation failed.
+
+::: moniker range=">= aspnetcore-2.1"
+
+For ASP.NET Core 2.1 or later web API controllers using the `[ApiController]` attribute, model validation occurs automatically. For more information, see [Automatic HTTP 400 responses](xref:web-api/index#automatic-http-400-responses).
```

Comment:
No "model validation _always_ occurs automatically". The difference with `[ApiController]` is the automatic 400 response containing details of issues when `ModelState.IsValid == false`.